React 18 peer dependency

# Bug report

## Packages affected

-   [ ] sandpack-client
-   [x] sandpack-react

## Description of the problem

The PR #375, which should had fixed the issue #459, set the React version to 18 (at least for the `@types` package), but did not update the version in the `peerDependencies` object which causes the already mentioned `ERESOLVE unable to resolve dependency tree` error (both for `react` and `react-dom` packages)

## What were you doing when the problem occurred?

Installing the `@codesandbox/sandpack-react` package

### What steps can we take to reproduce the problem?

Create and empty npm package with React 18 as dependency and then try to install the `@codesandbox/sandpack-react` package

### Your Environment

| Software                | Name/Version |
| ----------------------- | ------------ |
| Sandpack-client version |              |
| Sandpack-react version  |  1.2.2            |
| Browser                 |              |
| Operating System        |  Linux            |

--- a/sandpack-react/package.json
+++ b/sandpack-react/package.json
@@ -80,7 +80,7 @@
     "typescript": "4.4.4"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17",
-    "react-dom": "^16.8.0 || ^17"
+    "react": "^16.8.0 || ^17 || ^18",
+    "react-dom": "^16.8.0 || ^17 || ^18"
   }
 }

